pkg/tfbridge/tokens/tokens.go

```diff
@@ -62,3 +68,115 @@ func checkedApply[T comparable](dst *T, src T) {
 		*dst = src
 	}
 }
+
+// Describe the mapping from TF resource and datasource tokens to Pulumi resources and
+// datasources.
+type Strategy = info.Strategy
+
+// Assigns resource and data source tokens unless already specified by the user.
+//
+// Context: bridging a provider requires every TF resource and data source have a matching
+// entry in [Resources] and [DataSources]. Each entry needs to minimally specifying the
+// Pulumi token in [ResourceInfo.Tok]. The mapping entries can be done by hand for smaller
+// provides but become a chore for providers with 1000s of entries.
+//
+// ComputeTokens scans TF resources and datasources to populate missing entries as needed
+// with automatically computed Pulumi tokens.
+//
+// ComputeTokens always respects and does not modify pre-existing entires. The user can
+// therefore manually override the token decision by providing [ResourceInfo] or
+// [DataSourceInfo] entry prior to calling ComputeTokens.
+//
+// ComputeTokens respects [ProviderInfo.IgnoreMappings]. It will not create a mapping for
+// any token in [ProviderInfo.IgnoreMappings].
+func ComputeTokens(info *info.Provider, opts Strategy) error {
+	var errs multierror.Error
+
+	ignored := ignoredTokens(info)
+
+	err := computeDefaultResources(info, opts.Resource, ignored)
+	if err != nil {
+		errs.Errors = append(errs.Errors, fmt.Errorf("resources:\n%w", err))
+	}
+	err = computeDefaultDataSources(info, opts.DataSource, ignored)
+	if err != nil {
+		errs.Errors = append(errs.Errors, fmt.Errorf("datasources:\n%w", err))
+	}
+	return errs.ErrorOrNil()
+}
+
+type (
+	ResourceStrategy   = info.ResourceStrategy
+	DataSourceStrategy = info.DataSourceStrategy
+)
+
+func ignoredTokens(info *info.Provider) map[string]bool {
```

Comment:
duplicated above?

Reply:
This function was previously and still is duplicated. There was (and still is) a copy in `/pkg/tfbridge` and in `/pkg/tfbridge/tokens`.